### ISSUE TITLE
drop errnum parameter from flux_respond()

### ISSUE THREAD

**Issue by garlick** [flux-framework/flux-core#1539]
`flux_respond()` is used in RPC message handlers to construct a response message using the request message as a template:
```c
int flux_respond (flux_t *h, const flux_msg_t *request, int errnum, const char *s);
```
If _errnum_ is zero, the response includes a payload built from _s_.

If _errnum_ is non-zero, the response is constructed with no payload, ignoring _s_.

Some RPC handlers have been coded with the following pattern:
```c
  char *payload = NULL;
  int rc = -1;
  if (flux_request_decode (msg, ...) < 0)
    goto done;
  if (service_specific_function (..., &payload) < 0)
    goto done;
  rc = 0;
done:
  if (flux_respond (h, msg, rc < 0 ? errno : 0, payload) < 0)
    flux_log_error ("flux_respond");
  free (payload);
```
With the addition of `flux_respond_error()` in #1538, we can have a richer error response.  The `flux_respond_error()` call also, by its name, makes it more explicit that an error is occurring, and that whatever action was intended to be set in motion by the request ought to be halted and unwound.

RPC handlers could use the following pattern
```c
  char *payload = NULL;
  if (flux_request_decode (msg, ...) < 0
    goto error;
  if (service_specific_function (..., &payload) < 0)
    goto error;
  if (flux_respond (h, msg, 0, payload) < 0)
    flux_log_error ("flux_respond");
  free (payload);
  return;
error:
  if (flux_respond_error (h, msg, errno, NULL) < 0)
    flux_log_error ("flux_respond_error");
  free (payload);
```

I think I prefer the latter pattern even though it requires a duplication of cleanup code like the `free()` above.  My thought is that one would only make self-contained RPC handlers for very simple operations, and that more complex ones would use helpers like `service_specific_function()` above, which could minimize the amount of cleanup code that is duplicated.

What do people think?

If there is agreement we could drop the _errnum_ parameter from `flux_respond()` so that all error responses must use `flux_respond_error()`.

**Comment by chu11** [comment]
I think it's a good idea, more so just b/c of consistency.  ```flux_respond()``` doesn't allow the user to specify an error string.  So if the caller really does not want to send an error string, it should perhaps be explicit?

But skimming code, it appears no code currently takes advantage of sending the error string.  Is this something that we would anticipate being used more often in the future?  It does seem useful for debugging.  It'd be a pain to stick in error strings everywhere right now, but perhaps its a good refactoring step.

**Comment by garlick** [comment]
The ability to return an error string is brand new in #1538 so there aren't many users yet.  I guess the question is whether we want to change the many places where `flux_respond()` is used to return an error number to call `flux_respond_error()` instead.  In some places that means adding a conditional as in the first idiom shown above.

In addition to the need to change a lot of code, it's a question whether that idiom has merit and should continue to be possible.

I personally lean towards dropping the errnum param from `flux_respond()` but I could be easily convinced to leave things as they are if anybody thinks it would be a step backward.

**Comment by grondo** [comment]
Fixed way back in #2120
